server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -396,24 +398,46 @@ private void deleteSynonymsSetObjects(String synonymSetId, ActionListener<BulkBy
     }
 
     public void deleteSynonymsSet(String synonymSetId, ActionListener<SynonymsReloadResult<AcknowledgedResponse>> listener) {
-        deleteSynonymsSetObjects(synonymSetId, listener.delegateFailure((deleteObjectsListener, bulkByScrollResponse) -> {
-            if (bulkByScrollResponse.getDeleted() == 0) {
-                // If nothing was deleted, synonym set did not exist
-                deleteObjectsListener.onFailure(new ResourceNotFoundException("Synonym set [" + synonymSetId + "] not found"));
-                return;
-            }
-            final List<BulkItemResponse.Failure> bulkFailures = bulkByScrollResponse.getBulkFailures();
-            if (bulkFailures.isEmpty() == false) {
-                deleteObjectsListener.onFailure(
-                    new ElasticsearchException(
-                        "Error deleting synonym set: "
-                            + bulkFailures.stream().map(BulkItemResponse.Failure::getMessage).collect(Collectors.joining("\n"))
+
+        // Checks reloading the analyzer to understand usage. We can iterate on this solution to target the specific analyzer
+        reloadAnalyzers(synonymSetId, listener.delegateFailure((reloadListener, reloadResult) -> {
```

Comment:
let's instead of reloading analyzers, do the following:
- Add a parameter to `ReloadAnalyzersRequest` for validation only, for example `validateOnly=true`
- Pass this parameter upto [IndexAnalyzers:reload](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/analysis/IndexAnalyzers.java#L138) method.  
- If IndexAnalyzers:reload has this parameter as `true`, it doesn't do reloading, but returns only the list.

I think this could be quite fast change. WDYT?

Reply:
Thanks for clarifying - I'll make those changes tomorrow.

Reply:
I've implemented the changes, can you please take a look?